src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -925,35 +925,35 @@ public static int GetProcFSParentPid(int pid)
             }
 
             /// <summary>The native methods class.</summary>
-            internal static class NativeMethods
+            internal static partial class NativeMethods
             {
                 private const string psLib = "libpsl-native";
 
                 // Ansi is a misnomer, it is hardcoded to UTF-8 on Linux and macOS
                 // C bools are 1 byte and so must be marshaled as I1
 
-                [DllImport(psLib, CharSet = CharSet.Ansi)]
-                internal static extern int GetErrorCategory(int errno);
+                [LibraryImport(psLib, StringMarshalling = StringMarshalling.Utf8)]
+                internal static partial int GetErrorCategory(int errno);
 
-                [DllImport(psLib)]
-                internal static extern int GetPPid(int pid);
+                [LibraryImport(psLib)]
+                internal static partial int GetPPid(int pid);
 
-                [DllImport(psLib, CharSet = CharSet.Ansi, SetLastError = true)]
-                internal static extern int GetLinkCount([MarshalAs(UnmanagedType.LPStr)] string filePath, out int linkCount);
+                [LibraryImport(psLib, StringMarshalling = StringMarshalling.Utf8, SetLastError = true)]
+                internal static partial int GetLinkCount([MarshalAs(UnmanagedType.LPStr)] string filePath, out int linkCount);
```

Comment:
Do we really need [MarshalAs(UnmanagedType.LPStr)]?
https://github.com/dotnet/runtime/blob/82a2562fc9ee0986ee20ee309ad0bc259c561683/src/libraries/Common/src/Interop/Unix/System.Native/Interop.Stat.cs#L64

https://github.com/dotnet/runtime/blob/d320fd90a2976b388169dd51b3732ad01bcb4266/src/native/libs/System.Native/pal_io.c#L254

https://github.com/PowerShell/PowerShell-Native/blob/677c086f20209f70ecad379ec48de2562f1db668/src/libpsl-native/src/getlinkcount.cpp#L39

Reply:
This was just inherited from previous code, doesn't seem to be required, so removing.